lib/sparrow/apns/pool/supervisor.ex

```diff
@@ -0,0 +1,102 @@
+defmodule Sparrow.APNS.Pool.Supervisor do
+  @moduledoc """
+  Supervises a single APNS workers pool.
+  """
+  use Supervisor
+
+  @apns_dev_endpoint "api.development.push.apple.com"
+  @apns_prod_endpoint "api.push.apple.com"
+
+  @spec start_link(Keyword.t()) :: Supervisor.on_start()
+  def start_link(arg) do
+    Supervisor.start_link(__MODULE__, arg)
+  end
+
+  @spec init(Keyword.t()) ::
+          {:ok, {:supervisor.sup_flags(), [:supervisor.child_spec()]}}
+  def init(raw_apns_config) do
+    dev_raw_configs = Keyword.get(raw_apns_config, :dev, [])
+    prod_raw_configs = Keyword.get(raw_apns_config, :prod, [])
+
+    pool_configs =
+      (get_apns_pool_configs(dev_raw_configs, :dev) ++
+         get_apns_pool_configs(prod_raw_configs, :prod))
+      |> Enum.with_index()
+
+    children =
+      for {{{pool_config, pool_tags}, pool_type}, index} <- pool_configs do
+        id = String.to_atom("Sparrow.APNS.Pool.ID." <> Integer.to_string(index))
+
+        %{
+          id: id,
+          start:
+            {Sparrow.H2Worker.Pool, :start_link,
+             [pool_config, pool_type, pool_tags]}
+        }
+      end
+
+    Supervisor.init(children, strategy: :one_for_one)
+  end
+
+  @spec get_apns_pool_configs(Keyword.t(), :dev | :prod) :: [
+          {{Sparrow.H2Worker.Pool.Config.t(), [atom]}, :dev | :prod}
+        ]
+  defp get_apns_pool_configs(raw_pool_configs, pool_type) do
+    for raw_pool_config <- raw_pool_configs do
+      {get_apns_pool_config(raw_pool_config, pool_type), {:apns, pool_type}}
+    end
+  end
+
+  @spec get_apns_pool_config(Keyword.t(), :dev | :prod) ::
+          {Sparrow.H2Worker.Pool.Config.t(), [atom]}
+  defp get_apns_pool_config(raw_pool_config, pool_type) do
+    uri =
+      case pool_type do
+        :dev ->
+          Keyword.get(
+            raw_pool_config,
+            :endpoint,
+            @apns_dev_endpoint
```

Comment:
Since these two cases only differ in the default value, it might be cleaner to extract the default value alone and then call `Keyword.get(raw_pool_config, :endpoint, default_endpoint)`

Maybe even `Keyword.get(raw_pool_config, :endpoint, @apns_endpoint[pool_type])`

Reply:
seems legit :)